ci-templates/gitlab/image-builder.yml

```diff
@@ -0,0 +1,419 @@
+# SPDX-License-Identifier: Apache-2.0
+
+variables:
+  BASE:
+    value: "1"
+    description: "Build the base image?"
+  CAPELLA_BASE:
+    value: "1"
+    description: "Build the capella/base image?"
+  CAPELLA_CLI:
+    value: "1"
+    description: "Build the capella/cli image?"
+  CAPELLA_REMOTE:
+    value: "0"
+    description: "Build the capella/remote image?"
+  T4C_CLIENT_BASE:
+    value: "1"
+    description: "Build the t4c/client/base image?"
+  T4C_CLIENT_BACKUP:
+    value: "1"
+    description: "Build the t4c/client/backup image?"
+  T4C_CLIENT_EXPORTER:
+    value: "1"
+    description: "Build the t4c/client/exporter image?"
+  T4C_CLIENT_REMOTE:
+    value: "1"
+    description: "Build the t4c/client/remote image?"
+  T4C_CLIENT_REMOTE_PURE_VARIANTS:
+    value: "1"
+    description: "Build the t4c/client/remote/pure-variants image?"
+  CAPELLA_EASE:
+    value: "1"
+    description: "Build the capella/ease image?"
+  CAPELLA_EASE_REMOTE:
+    value: "1"
+    description: "Build the capella/ease/remote image?"
+  CAPELLA_READONLY:
+    value: "1"
+    description: "Build the capella/readonly image?"
+  T4C_CLIENT_EASE:
+    value: "0"
+    description: "Build the t4c/client/ease image?"
+  T4C_CLIENT_EASE_REMOTE:
+    value: "0"
+    description: "Build the t4c/client/ease/remote image?"
+  T4C_CLIENT_EASE_REMOTE_DEBUG:
+    value: "0"
+    description: "Build the t4c/client/ease/remote/debug image?"
+  CAPELLA_DOCKER_IMAGES_REVISION:
+    value: "main"
+    description: "Revision of the Capella Docker images Github repository (https://github.com/DSD-DBS/capella-dockerimages)"
+  CAPELLA_VERSION:
+    value: "6.0.0"
+    description: "Capella version. Please make sure that a subdirectory with the name of the value exists. The value must be valid ASCII and may contain lowercase and uppercase letters, digits, underscores, periods and dashes."
+  DOCKER_BUILD_ARGS: "--no-cache"
+  BUILD_ARCHITECTURE: amd64
+
+default:
+  image: $DOCKER_REGISTRY/base
+  tags:
+    - docker
+  before_script:
+    - &normalizeTag export DOCKER_TAG=$CAPELLA_VERSION-$(echo $CAPELLA_DOCKER_IMAGES_REVISION | sed 's/[^a-zA-Z0-9.]/-/g')-$CI_COMMIT_REF_NAME
+    - export BASE_IMAGE=$BASE_IMAGE:$DOCKER_TAG
+
+.github: &github
+  - git clone --recurse-submodules https://github.com/DSD-DBS/capella-dockerimages.git
+  - cd capella-dockerimages
+  - git checkout $CAPELLA_DOCKER_IMAGES_REVISION
+
+.docker: &docker
+  - docker info
+  - docker login -u $DOCKER_REGISTRY_USER -p $DOCKER_REGISTRY_PASSWORD $DOCKER_REGISTRY
+  - docker pull $BASE_IMAGE
+
+.push: &push
+  - docker push $IMAGE:$DOCKER_TAG
+
+.ease: &ease
+  - cp -R ../ease/extensions/* ease/extensions/
+
+.prepare: &prepare
+  - cd $CAPELLA_VERSION
+  - *github
+
+base:
+  stage: build
+  needs: []
+  rules:
+    - if: '$BASE == "1"'
+      when: always
+  before_script:
+    - *normalizeTag
+  variables:
+    BASE_IMAGE: $DOCKER_REGISTRY/base
```

Comment:
We should change this here to a generic base image (debian) otherwise you can't use the ci template when no base image exists in the registry. Would be good to add 1-2 sentences in the documentation about that and how to set your own base image.

Reply:
I've changed the base image, but I'd not add a section in the README. Basically every value in the `.gitlab-ci.yml` can be overwritten using the typical gitlab ci syntax for overwriting templates.